sql/hive-shaded/pom.xml

```diff
@@ -0,0 +1,252 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one or more
+  ~ contributor license agreements.  See the NOTICE file distributed with
+  ~ this work for additional information regarding copyright ownership.
+  ~ The ASF licenses this file to You under the Apache License, Version 2.0
+  ~ (the "License"); you may not use this file except in compliance with
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~    http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.apache.spark</groupId>
+    <artifactId>spark-parent_2.12</artifactId>
+    <version>3.3.0-SNAPSHOT</version>
+    <relativePath>../../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>spark-hive-shaded_2.12</artifactId>
+  <packaging>jar</packaging>
+  <name>Spark Project Hive Shaded</name>
+  <url>http://spark.apache.org/</url>
+  <properties>
+    <sbt.project.name>hive-shaded</sbt.project.name>
+  </properties>
+
+  <!-- dependencies in the following are optional because we don't want to leak them as
+    transitive dependencies, as they are already included in the shaded jar -->
```

Comment:
After https://github.com/apache/spark/pull/34085 lands I think we'll be able to omit the `<optional>`s here. I previously wasn't familiar with Maven's `<optional>` dependencies, so I spotted that `createDependencyReducedPom` issue while trying to understand why they were needed in this PR.

Reply:
Nice catch! yes I think we can remove these now (after the follow-up is also done).